cmd/service-controller/controller.go

```diff
@@ -930,6 +930,25 @@ func (c *Controller) updateServiceBindings(required types.ServiceInterface, port
 			delete(c.bindings, required.Address)
 			return nil
 		}
+		ports := bindings.GetIngressPorts()
+		if len(ports) < len(required.Ports) {
```

Comment:
If we created an interface for nextFreePort/release then this updating logic could be moved into the binding itself which might be a little neater overall. That could also be a separate refactoring however.